app/cloud.go

```diff
@@ -62,57 +121,158 @@ func (a *App) NotifySystemAdminsToUpgrade(c *request.Context, currentUserTeamID
+	}
+
+	return false
+}
 
-	pref := model.Preference{
-		UserId:   fakeId, // to only have one preference for now and not a preference per user
-		Category: model.PreferenceCloudUserEphemeralInfo,
-		Name:     model.CloudNotifyAdminInfo,
-		Value:    string(out),
+func (a *App) CanNotifyAdmin(trial bool) bool {
+	systemVarName := lastUpgradeNotificationTimeStamp
+	if trial {
+		systemVarName = lastTrialNotificationTimeStamp
 	}
 
-	if err := a.Srv().Store.Preference().Save(model.Preferences{pref}); err != nil {
-		mlog.Warn("Encountered error saving cloud_user_ephemeral_info preference", mlog.Err(err))
+	sysVal, sysValErr := a.Srv().Store.System().GetByName(systemVarName)
+	if sysValErr != nil {
+		var nfErr *store.ErrNotFound
+		if errors.As(sysValErr, &nfErr) { // if no timestamps have been recorded before, system is free to notify
+			return true
+		}
+		mlog.Error("Cannot notify", mlog.Err(sysValErr))
+		return false
 	}
 
-	return nil
+	lastNotificationTimestamp, err := strconv.ParseFloat(sysVal.Value, 64)
+	if err != nil {
+		mlog.Error("Cannot notify", mlog.Err(err))
+		return false
+	}
+
+	coolOffPeriodDaysEnv := os.Getenv("MM_CLOUD_NOTIFY_ADMIN_COOL_OFF_DAYS")
+	coolOffPeriodDays, parseError := strconv.ParseFloat(coolOffPeriodDaysEnv, 64)
+	if parseError != nil {
+		coolOffPeriodDays = defaultCloudNotifyAdminCoolOffDays
+	}
+	daysToMillis := coolOffPeriodDays * 24 * 60 * 60 * 1000
+	timeDiff := model.GetMillis() - int64(lastNotificationTimestamp)
+	return timeDiff >= int64(daysToMillis)
 }
 
-// Ensure cloud service wrapper implements `product.CloudService`
-var _ product.CloudService = (*cloudWrapper)(nil)
+func (a *App) FinishSendAdminNotifyPost(trial bool, now int64) {
+	systemVarName := lastUpgradeNotificationTimeStamp
+	if trial {
+		systemVarName = lastTrialNotificationTimeStamp
+	}
 
-// cloudWrapper provides an implementation of `product.CloudService` for use by products.
-type cloudWrapper struct {
-	cloud einterfaces.CloudInterface
+	val := strconv.FormatInt(model.GetMillis(), 10)
+	sysVar := &model.System{Name: systemVarName, Value: val}
+	if err := a.Srv().Store.System().SaveOrUpdate(sysVar); err != nil {
+		mlog.Error("Unable to finish send admin notify post job", mlog.Err(err))
+	}
+
+	// all the notifications are now sent in a post and can safely be removed
+	if err := a.Srv().Store.NotifyAdmin().DeleteBefore(trial, now); err != nil {
+		mlog.Error("Unable to finish send admin notify post job", mlog.Err(err))
+	}
+
+}
+
+func (a *App) groupNotifyAdminByUser(data []*model.NotifyAdminData) map[string][]*model.NotifyAdminData {
+	myMap := make(map[string][]*model.NotifyAdminData)
+	for _, d := range data {
+		myMap[d.UserId] = append(myMap[d.UserId], d)
+	}
+
+	return myMap
+}
+
+func (a *App) groupNotifyAdminByFeature(data []*model.NotifyAdminData, T i18n.TranslateFunc) map[model.MattermostPaidFeatureTranslationId][]*model.NotifyAdminData {
+	T("mattermost.feature.guest_accounts")
```

Comment:
@neallred I feel like there's a better way to keep the translation ids in the en.json file. But not sure how. Could this suffice?